### PR TITLE
Adjusted stage transition rules to make stages less likely to roll back

### PR DESCRIPTION
**Pull request by ZhixiaoSu**
It has been too easy for animals on the curriculum uncoupled_no_baiting_2p2rwdDelay159.py to roll back to earlier stages. I adjusted 'roll back' rules to reduce the probability.
Main changes: 
1. Forced animals to stay on Stage 3 for at least 3 days before rolling back, and lowered criteria to stay in stage 3
2. At Stage Final, lowered the criteria to stay in Stage Final to account for day to day variabilities.
3. At Stage 2 increased no lick delay to 0.25 to avoid sharp performance drop at Stage 3.

Here's the parameter plot and transition graph: 
Uncoupled Without Baiting_curriculum_v2 2rwdDelay159_schema_v1 0_paras
Uncoupled Without Baiting_curriculum_v2 2rwdDelay159_schema_v1 0_rules

### CODE REVIEW

File: code/aind_auto_train/curriculums/uncoupled_no_baiting_2p2rwdDelay159.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 # Note this could be any string, not necessarily one of the Task enums
 curriculum_name = Task.C0B0
 curriculum_version = "2.2rwdDelay159"
-curriculum_description = '''2024-06-18 less rollback from FINAL; more AutoWater; AutoIgnore=25'''
+curriculum_description = '''2024-06-18 less rollback from FINAL; more AutoWater; AutoIgnore=25, 2024-06-18 less rollback from stage3'''
```
Comment (hanhou) [inline code/aind_auto_train/curriculums/uncoupled_no_baiting_2p2rwdDelay159.py:27]:
```suggestion
curriculum_description = '''2024-06-18 less rollback from FINAL; more AutoWater; AutoIgnore=25, 2024-07-01 less rollback from stage3'''
```

File: code/aind_auto_train/curriculums/uncoupled_no_baiting_2p2rwdDelay159.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 # Note this could be any string, not necessarily one of the Task enums
 curriculum_name = Task.C0B0
 curriculum_version = "2.2rwdDelay159"
```
Comment (hanhou) [inline code/aind_auto_train/curriculums/uncoupled_no_baiting_2p2rwdDelay159.py:26]: Do you think it's better to use a new curriculum version like 2.3?

Comment (ZhixiaoSu) [reply code/aind_auto_train/curriculums/uncoupled_no_baiting_2p2rwdDelay159.py:26]: I'm debating between 2.2.1 and 2.3 actually. Compared to 2.1, 2.2 should have higher autowater and less likely rolling back, but the rwddelay curriculum in some way skipped this change. Maybe it is better to be 2.2.1 to be consistent with other curriculums?

Comment (ZhixiaoSu) [reply code/aind_auto_train/curriculums/uncoupled_no_baiting_2p2rwdDelay159.py]: I reverted changes in original 2.2 and added a new 2.2.1 curriculum. It's called uncoupled_no_baiting_2p2p1rwdDelay159.py.

**Pull request merged**